crates/core/tedge_api/src/entity_store.rs

```diff
@@ -97,13 +98,27 @@ impl EntityStore {
         self.get(mqtt_id)
     }
 
+    /// Returns the entity attached to a topic, if any
+    pub fn get_entity_from_topic(&self, topic: &Topic) -> Option<&EntityMetadata> {
+        let entity_topic = EntityTopic::try_from(topic.clone()).ok()?;
+        self.get(entity_topic.entity_id())
+    }
+
     /// Returns the MQTT identifier of the main device.
     ///
     /// The main device is an entity with `@type: "device"`.
     pub fn main_device(&self) -> EntityIdRef {
         self.main_device.as_str()
     }
 
+    /// Returns the name of main device.
+    pub fn main_device_name(&self) -> &str {
```

Comment:
> But probably 99% this is what we want, so `main_device` could be changed to just return `EntityMetadata`.

Yes, that would be better. Especially, because `entity_store.get(entity_store.main_device())` returns an `Option` despite there is always some main device.

> One problem is that `EntityMetadata` doesn't contain entity MQTT identifier, but that can perhaps be added later in a way that doesn't just duplicate the string

Yes, this will also be really convenient.